amundsen_application/static/js/components/DashboardPage/ImagePreview/index.spec.tsx

```diff
@@ -1,6 +1,7 @@
 import * as React from 'react';
+import { Modal } from 'react-bootstrap';
 
-import { shallow } from 'enzyme';
+import { mount } from 'enzyme';
```

Comment:
Oh, why `mount`? The initial approach we took for this repo was to use `shallow` rendering unless we had a case that justified using `mount`. We were concerned about how expensive it would be to call `mount`, and how long it could take to run tests as the repository grew. 

Is it that `mount` allows for your preferred style of testing in the `lifecycle` section below? I'm not opposed to using `mount` where it has a benefit, but want to make sure I understand the benefit.

Reply:
Yes, mount was necessary in order for the lyfecycle section to work, as it is rendering a child component (PreviewModal) and simulating events on it.

This was one of the things that surprised me at the beginning, as in my previous teams we never used shallow. It has some bad press, and even if it has improved over time (time ago you couldn't call some methods), it gives a false sense of security.

You are right about mount being slower, but it is a matter of milliseconds. Also, having to switch between one and the other depending on if you have nested components is a bit painful.

Reply:
Here is a more opinionated response to the mount vs shallow: https://kentcdodds.com/blog/why-i-never-use-shallow-rendering

He has his own way, so take it with a grain of salt :)

Reply:
I've read that article previously, along with others that basically take the opposite standpoint. I'll invest more research here at some point to be able to provide clearer directions for newcomers to the repository. Essentially even with all the pros and cons available in various posts and documents there's some effort on my part to revisit how much the application and components have changed since the initial decision, in order to try to draw some clearer connections between other's opinions and the needs of the app.

Internally our FE working group has actually been working on some more official guidance for unit testing -- that doc is almost finalized though and I've been waiting on that or guidance on improving our testing practices as well. I'll send you the link. (fyi: They do happen to lean towards `react-testing-library` and no shallow rendering.)

Reply:
Totally, it makes sense to have a clear idea of where to go before jumping into refactoring tests!

I will check that doc, I received the email this week and added it to my reading list, I will prioritize it now. 

To be honest, I am not so much into the React Testing Library, I think enzyme is fine for most of the cases if used properly. That said, I don't know about hooks, maybe that changes everything.

Thanks!